Automattic/jetpack
# Social: Add SIG preview component

This change introduces a new component that renders the current preview for SIG. It goes to the API endpoint that was introduced in #29596, which returns the token.

## Proposed changes:

* Created new component
  * The image URL is updated each time a needed property changes
  * String fields are debounced to 1500ms, to decrease calls.
* Added component tests for the new component

### Other information:


## Jetpack product discussion


1204116019981263-as-1204165892483477
## Does this pull request change what data or activity we track or use?


No.
## Testing instructions:






* Checkout this PR, buy the Advanced Plan.
* Go to new post
* In the editor SIG panel, you should see the default generated preview
* Change the title, the title should change on the image after 1500ms.
* Set a featured image, it should show behind the title
* Set a custom message, that should overwrite the title on the image. While having a custom message, changes to the title should not cause any API calls.
* Pick a custom image, that should overwrite the featured image.
